File: VB/GridBeforePrint/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing

Namespace GridBeforePrint

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Private customerList As List(Of Customer)

        Private Function GetData() As List(Of Customer)
            Dim list As List(Of Customer) = New List(Of Customer)()
            For i As Integer = 0 To 15 - 1
                Dim customer As Customer = New Customer() With {.FirstName = String.Format("FirstName {0}", i), .LastName = String.Format("LastName {0}", i), .Id = i}
                list.Add(customer)
            Next

            Return list
        End Function

        Public Sub New()
            InitializeComponent()
            myGridView1.OptionsView.ShowPreview = True
            myGridView1.OptionsView.AutoCalcPreviewLineCount = False
            myGridView1.PreviewFieldName = "FirstName"
            customerList = GetData()
            myGridControl1.DataSource = customerList
            Me.Controls.Add(myGridControl1)
            AddHandler myGridView1.HeaderPrintEvent, AddressOf MyGridView1_HeaderPrintEvent
            AddHandler myGridView1.SamplePrintEvent, AddressOf MyGridView1_SamplePrintEvent
        End Sub

        Private Sub MyGridView1_SamplePrintEvent(ByVal sender As Object, ByVal args As SamplePrintEventArgs)
            args.Brick.Style.BackColor = Color.Pink
        End Sub

        Private Sub MyGridView1_HeaderPrintEvent(ByVal sender As Object, ByVal args As HeaderPrintEventArgs)
            args.Brick.Style.BackColor = Color.PowderBlue
        End Sub

        Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs)
            myGridControl1.ShowPrintPreview()
        End Sub
    End Class
End Namespace

File: CS/GridBeforePrint/Form1.cs
using System;
using System.Collections.Generic;
using System.Drawing;

namespace GridBeforePrint
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        private List<Customer> customerList;

        private List<Customer> GetData()
        {
            List<Customer> list = new List<Customer>();
            for (int i = 0; i < 15; i++)
            {
                Customer customer = new Customer()
                {
                    FirstName = string.Format("FirstName {0}", i),
                    LastName = string.Format("LastName {0}", i),
                    Id = i
                };
                list.Add(customer);
            }
            return list;
        }

        public Form1()
        {
            InitializeComponent();
            myGridView1.OptionsView.ShowPreview = true;
            myGridView1.OptionsView.AutoCalcPreviewLineCount = false;
            myGridView1.PreviewFieldName = "FirstName";
            customerList = GetData();
            myGridControl1.DataSource = customerList;
            this.Controls.Add(myGridControl1);

            myGridView1.HeaderPrintEvent += MyGridView1_HeaderPrintEvent;
            myGridView1.SamplePrintEvent += MyGridView1_SamplePrintEvent;
        }

        private void MyGridView1_SamplePrintEvent(object sender, SamplePrintEventArgs args)
        {
            args.Brick.Style.BackColor = Color.Pink;
        }

        private void MyGridView1_HeaderPrintEvent(object sender, HeaderPrintEventArgs args)
        {
            args.Brick.Style.BackColor = Color.PowderBlue;
        }

        private void simpleButton2_Click(object sender, EventArgs e)
        {
            myGridControl1.ShowPrintPreview();
        }
    }
}
